Hydrophonics Stuff I guess

sunshinehunter opened the pull request:
Makes #7919 "nerfs" less unforgiving, and fix some minor chat spam, cool

:cl:
tweak: DNA manipulator has some more uses with earlier machine part tiers, half the amount of earthsblood/ambrosia is needed to make hydrophonic trays self-sufficent
/:cl:

sunshinehunter commented:
I have no idea what happened with paradise.dme, by the way...

Crazylemon64 commented:
`git checkout HEAD~1 paradise.dme`
`git add paradise.dme`
`git commit --amend`
`git push origin master`

Also you commited this to your master branch, dishonor upon your family

sunshinehunter commented:
Oh dear.

![What?](http://i0.kym-cdn.com/photos/images/original/000/234/739/fa5.jpg)

taukausanake commented:
Separate branches also mean you can have dank meme branch names

Birdtalon commented:
It's been less than a week since the values for Hydroponics was changed. Not nearly enough time for a sensible review in my opinion.

FalseIncarnate commented:
The age adjustments were put in place because TG's botany is MUCH faster paced than what many of our botany players were used to, since TG (at least at the time, may have changed) didn't average 90+ minute rounds. That lead to newer botanists being overwhelmed with a high-paced set of mechanics (which as a 0-day job was not ideal) and experienced botanists being able to quickly overwhelm the chef's rate of produce usage/burning with little to no modifications early in the round.

The aging rate slowed things back down to closer to how they were pre-TG botany, making the job more relaxed for newer players and giving experienced players more time playing with genes than frantically rushing to keep all the trays alive.

It also meant things like ambrosia gaia took a bit longer to obtain, since before the aging tweak, I'd noticed botany filled with gaia trays within the first 10-15 minutes of a shift... which, for something that was meant to be difficult to obtain, kinda made the entire job moot.

sunshinehunter commented on code/modules/hydroponics/hydroponics.dm:
```diff
@@ -551,10 +545,10 @@
 		adjustWater(round(S.get_reagent_amount("fishwater") * 1))
 
-	// Ambrosia Gaia produces earthsblood.
+	// You know what else has /datum/reagents, besides a beaker with earthsblood in it? That's right, ambrosia gaia!
```
Despite saying so in #7919, Ambrosia Gaia composting is still a thing, since it contains /datum/reagents.

Since the grinder rounds up the reagents from the ingredients, using the grinder is FAR more viable. Example: Have a gaia branch with 1.4 earthsblood. Grinding gives 2 earthsblood. Now the question is, declare it as ""unintended mechanics depth"" or do something about it, like actually make gaia not compostable. You can still compost other plants though, and modify them to produce earthsblood. So, what the hell was the point of making gaia **NOT compostable**??

sunshinehunter commented:
@Birdtalon In my opinion it is, since not much changed besides the changes for the DNA manipulator and making self-sustainable trays.

@FalseIncarnate I realized that too, which is why for now don't consider it anymore

sunshinehunter commented:
This PR is now ready for merge.

tigercat2000 commented:
Missing changelog. As a reminder, the format is
```
:cl:
add/del/fix/tweak: Description goes here
add/del/fix/tweak: Description 2 goes here
/:cl:
```

sunshinehunter commented:
_Now_ it is ready!

ktoma36 commented:
Mmmm... I think a month is a good amount of time to see the effects of the new botany. Is this PR getting snuffed out?

Crazylemon64 commented on code/modules/hydroponics/gene_modder.dm:
```diff
@@ -43,25 +43,31 @@
-			max_potency = initial(max_potency) + (M.rating**3) // 51,58,77,95 	 Clamps at 100
+		// 50,65,80,95
+		// Max natural potency is 100, machine clamps at 95
+		max_potency = min(initial(max_potency) + M.rating * 15, 95)
```
Linear scaling seems like a better idea than cubic scaling, good choice

Crazylemon64 commented on code/modules/hydroponics/gene_modder.dm:
```diff
@@ -17,10 +17,10 @@
-	var/min_wchance = 67
-	var/min_wrate = 10
+	var/min_wrate = 8
+	var/min_wchance = 24
```
The weed chance being dropped from 67 to 24 is quite drastic

ktoma36 replied:
Isn't that the percentage chance for a plant to get overran with weeds every cycle once there are too many weeds in the tray? A one in for chance of that happening is still pretty high especially since production rate is capped at 9 for the machine.

sunshinehunter replied:
An over 50% weed chance is way too much, players shouldn't be punished for having bad luck with mutations and then needing to stick with it, if they don't recieve upgrades (mutations like to give plants 67% weed chance, which is frustrating if you can't change it in further generations of the plant)

Crazylemon64 replied:
Mmm, seems fair

Crazylemon64 commented on code/modules/hydroponics/gene_modder.dm:
```diff
@@ -43,25 +43,31 @@
-			min_production = 12 - (SM.rating * 3) //9,6,3,1. Requires if to avoid going below clamp [1]
+		// 7,5,3,1
+		// Min natural production is 1
+		min_production = max(initial(min_production) - SM.rating * 2, 1)
```
Not sure about dropping base production time by 2

sunshinehunter replied:
Production and the growing cycle time is linear. A production value of 7 is still pretty insignificant, 9 is just completely useless, so this value doesn't matter much at Tier 1 parts.

ktoma36 commented:
I can say from experience with the current way botany is you cannot leave your plants unattended for over ten minutes. I got welder bombed while getting some stuff from chemistry one round. In the 15 min it took for me to get cloned *all* my plants were dead. This is with each plant having weed adaption as well. Maybe changing the max endurance would be a good idea as well, since having plants take longer to produce harvest combined with a short lifespan really sucks to deal with.

sunshinehunter commented:
The current requested changes are not needed if no proper reason is stated, as they are just nitpicking in my opinion, and there doesn't seem much of a discussion about the change anyway. In the end, the weed chance drop is mostly for people who don't want to remove weeds every 15 seconds from their overly mutated plants so they don't die instantly, and the base production time decrease for Tier 1 (9->7) and Tier 2 (6->5) is irrelevant too, since that's just a few cycles we are talking about.

sunshinehunter commented:
Changed the changelog to be more in-depth

sunshinehunter commented:
Are any changes still requested? This PR has been ready for a month already, without any changes.

ktoma36 commented:
Its been 42 days since this PR has been put up and the author has been keeping with it for the whole time. 

@Crazylemon64 It still says you are requesting changes

sunshinehunter commented:
Yup, it's pretty disappointing

Fox-McCloud commented:
Following a further analysis of the code, as well as a discussion between the Heads of Staff and Maintainers, it was decided to close this PR; not only does it more or less undo the previous change to an unnecessarily excessive degree, but it does so via numerous undocumented changes on how plants and plant growth operate, which is simply not acceptable.

sunshinehunter commented:
This PR doesn't revert any changes. I tweaked the values for the machine part tiers, documented in the code, and cleaned some of the relevant code up.

The plant growth time doesn't change, I removed the unnecessary `#define HYDRO_CYCLES_PER_AGE 2` and doubled the cycle delay, which changes nothing in effect, the plants still grow at the same speed.

It would have been great if any of these concerns would have been named before you closed the PR over a month after it was finished.
